Update FST Online QA Plots

zyzhang-0 opened the pull request:
In the Shift QA plots, add:
MaxTB by APV;
TB(1)/TB(0-2) fraction by APV.

zyzhang-0 committed:
Update FST Online QA Plots

genevb commented:
Hi, @zyzhang-0 and @jml985 
Do you know if it is critical to deploy this in DEV quickly (i.e. does this need to get past the git review process immediately)? I'm guessing Jeff updates the Jevp codes used at the experiment quickly separately from the git review, but it would be good to be clear about it.

zyzhang-0 commented:
> Hi, @zyzhang-0 and @jml985 Do you know if it is critical to deploy this in DEV quickly (i.e. does this need to get past the git review process immediately)? I'm guessing Jeff updates the Jevp codes used at the experiment quickly separately from the git review, but it would be good to be clear about it.

Hello @genevb ,
Yes it is. I've talked to @jml985 and he will take care of it this afternoon or evening, after he and I are on the same page of the layout.

zyzhang-0 committed:
Merge branch 'star-bnl:main' into main

zyzhang-0 committed:
Jevp production Sync (#1)

* testing

* testing

* test git

* test git

* updated again

* update

* update

* Tonko added TONKO_TRASH

* Modified yada

* moew

* updated yada

* more

* More tests

* updates for root6

* fixed typo

* blah

* blah2

* yeah

* update readers for STGC and FCS

* rollback TRG_FCS changes do to protype changes

* and delete the files added to finish rollback

* removed unneeded comments, and test files

* got rid of comment

* Initial Implement of FST Online Monitor

* update 2021 trigger data format to version 0x47

* Initial Implement of FST Online Monitor (#169)

* Initial Implement of FST Online Monitor

* Update OnlTools/Jevp/StJevpBuilders/fstBuilder.h

Co-authored-by: Dmitri Smirnov <dmixsmi@gmail.com>

* Adding StTriggerData2022 for new v47 trigger data structure

* test git

* Tonko added TONKO_TRASH

* Modified yada

* moew

* updated yada

* more

* More tests

* update readers for STGC and FCS

* rollback TRG_FCS changes do to protype changes

* and delete the files added to finish rollback

* removed unneeded comments, and test files

* update 2021 trigger data format to version 0x47

* added server for fstBuilder

* Use 64 bit detector presence flag to include STGC and FTS

* Clean up launch process

* fstBuilder with two input ped files

* fixed bug for CMN plots

* forward declaration daqReader

* update etofBuilder

* use a static base store

* fix display issue

* Delete script after use

* add

* change hitCut to 3.0

* fixed geoIdx bug

* git rid of eemcBuilder hack that was leading to seg faults.  Improved logging

* add zoomed plot

* sTGC Online plots for full detector with VMM

* stgc updats

* STGC: Add timing and charge plots, outline detectors.

* update FST monitor plots

* added logging, increased time delay for writing plots

* added logging, increased time delay for writing plots

* Update sTGC plots, show diagonal strips, fit trigger time delay

* Add online TOT histograms for MTD

* Update code in StRoot/StLaserAnalysisMaker (#174)

* Geometry update for 2022 (#173)

* Geometry update for 2022

- Implement first cut of the y2022 geometry @ timestamp 20211010.000000
-- sTGC and forward silicon detectors are still in development
-- overlaps are a concern for the initial geometries
-- expecting geometry updates end of month timescale

- Remove deprecated dev2021 geometry model
- Advance dev2022 timestamp by 100 years to not interfere

* Restore dev2021 @ advanced timestamp / refine comments

* And add back in the dev2021 tag to xgeometry

* And get the y2022 declaration correct (time stamp, tags...)

* Update CHANGELOG (#119)

* CHANGELOG: Update SL21c_0 release details

* CHANGELOG: Add SL21c_2 release details

* CHANGELOG: Add SL21c_3 release details

* CHANGELOG: Add SL21c_4 release details

* CHANGELOG: Add SL21c_5 release details

* CHANGELOG: Update future release draft

[skip ci]

* Add Anti-H4 Lambda decay to Anti-He4 + pion- (#176)

* Add H3LambdaBar decay to gstar library

* Add H4LambdaBar, He4LambdaBar & He5LambdaBar hypernuclei to StarClassLibrary.
Add only H4LambdaBar to the particle table (others are placeholders assuming
a request is made at some point).

* fixed array size (#178)

Co-authored-by: Dmitry Arkhipkin <dmitry@localhost.localdomain>

* D edx fxt2019 (#175)

* Merge with TFG

* Add defaults

* Add new chairs

* Add table for Gating Grid time shape

* Add pulls with Predictions depneded (old) and non depended (new) on log_2{<dX>)

* New schema for Run XIX and XX fixed target: use GatingGrid time shape & TpcZCorrectionC, absorb log_2(dX) dependence in TpcLengthCorrectionMD2

* Disable IntegrateAdc

* Averaged magnet current to set magnetic field strength (#170)

* Add Gating Grid time shape

* Add new options

* Switch to averaged STAR main magnet current, add a few new chairs

* Xianglei Zhu, 09/11/2021: FXT3p85_2018 and 09/14/2021: Isobar parameters (#156)

* Xianglei Zhu, 09/11/2021: FXT3p85_2018 and 09/14/2021: Isobar parameters parameters

* Update StarDb/Calibrations/tpc/TpcResponseSimulator.20171220.000001.C

Co-authored-by: Dmitri Smirnov <dmixsmi@gmail.com>

* Introduce FXT chain option to enable anything FXT-specific (#161)

Moving forward.

* Add fix for fixed Target run with broken yellowIntensity information (#180)

* Changing FCS Wcal and Hcal x positions in Geometry file based on run 21 survey (#177)

* Changing FCS Wcal and Hcal x positions in Geometry file beased on run21 survey
Also updating macro to store those new numbers in run22 offline DB initialization

* Apply suggestions from code review

Co-authored-by: Dmitri Smirnov <dmixsmi@gmail.com>

Co-authored-by: Dmitri Smirnov <dmixsmi@gmail.com>

* Add good event flag to ETOF headers and StPicoEvent (#160)

* fixed 64b integer conversion in getField() method and added additional setters from etofIn001 version of this file. Needed to unpack Get4Status bit into event files

* merged with group development to have support for local run-by-run calibrations in the repository version. Nessecary only for creating calibrations, but should be available in the offical code. Added calculation of goodEventFlag for every counter from Get4 missmatch bits and pulser digis.

* merged with group development to have support for local run-by-run calibrations in the repository version. Nessecary only for creating calibrations, but should be available in the offical code. Added calculation of goodEventFlag for every counter from Get4 missmatch bits and pulser digis.

* <strstring.h> header include needed to be added for local compilation (?)

* added eTofGoodEventFlag to mark counters which are good foer analysis in each event

* moved struct to public to avoid root6 issue. Changed class dev version back to 0

* added kStFatal return in ::init if calib histo files are corrupt

* Changed class dev version back to 0

* Trying to fix merger

* Trying to fix merger

* added eTofGoodEventFlag to mark counters which are good foer analysis in each event

* moved struct to public to avoid root6 issue. Changed class dev version back to 0

* added kStFatal return in ::init if calib histo files are corrupt

* Changed class dev version back to 0

* decremented classDef version of StEtofCollection. Makers compile correct header version after rebase without increment

* removed double line in MuEtofHeader constructor

* included eTOF constants header and moved constants here

* included eTOF constants header and moved constants here

* added constant nbGet4s here for use in event headers

* Removed strstr include in StMuDstMaker.cxx. Now indentical to repository version

* Moved vector init in eTOF headers to initializer list

* Changed default reference pulser setting in calib maker back to data base. Decremented etof collection ClassDef for real this time

Co-authored-by: Philipp Weidenkaff <weidenkaff@rcas6005.rcf.bnl.gov>
Co-authored-by: PhilippWeidenkaff <weidenkaff@physi.uni-heildeberg.de>

* Adding 3 access functions for revTick (#183)

* Move histogram creation of reference pulser outside of daQA flag (#182)

* fixed 64b integer conversion in getField() method and added additional setters from etofIn001 version of this file. Needed to unpack Get4Status bit into event files

* merged with group development to have support for local run-by-run calibrations in the repository version. Nessecary only for creating calibrations, but should be available in the offical code. Added calculation of goodEventFlag for every counter from Get4 missmatch bits and pulser digis.

* merged with group development to have support for local run-by-run calibrations in the repository version. Nessecary only for creating calibrations, but should be available in the offical code. Added calculation of goodEventFlag for every counter from Get4 missmatch bits and pulser digis.

* <strstring.h> header include needed to be added for local compilation (?)

* added eTofGoodEventFlag to mark counters which are good foer analysis in each event

* moved struct to public to avoid root6 issue. Changed class dev version back to 0

* added kStFatal return in ::init if calib histo files are corrupt

* Changed class dev version back to 0

* Trying to fix merger

* Trying to fix merger

* added eTofGoodEventFlag to mark counters which are good foer analysis in each event

* moved struct to public to avoid root6 issue. Changed class dev version back to 0

* added kStFatal return in ::init if calib histo files are corrupt

* Changed class dev version back to 0

* decremented classDef version of StEtofCollection. Makers compile correct header version after rebase without increment

* removed double line in MuEtofHeader constructor

* included eTOF constants header and moved constants here

* included eTOF constants header and moved constants here

* added constant nbGet4s here for use in event headers

* Removed strstr include in StMuDstMaker.cxx. Now indentical to repository version

* Moved vector init in eTOF headers to initializer list

* Changed default reference pulser setting in calib maker back to data base. Decremented etof collection ClassDef for real this time

* Moved reference pulser histogram creation outside of doQA flag. Fixing Issue #181

Co-authored-by: Philipp Weidenkaff <weidenkaff@rcas6005.rcf.bnl.gov>
Co-authored-by: PhilippWeidenkaff <weidenkaff@physi.uni-heildeberg.de>

* Updates	from Tonko on StRoot/RTS/src/DAQ_FCS and TRG_FCS (#179)

* Updates	from Tonko on StRoot/RTS/src/DAQ_FCS and TRG_FCS
Adding run22 trigger algo in StRoot/RTS/src/TRG_FCS
StFcsDb updates on flexible ETGain, fixing reading Gains from text, macro for gain DB
StFcsWaveformFitMaker adding stage0 peak finder	QA
StFcsTriggerSimMaker adding run22 TCU emulator,	reading	thresholds from text, ET/E gain match selection

* Adding forgotton Stage2 algo codes

* Adding EPD for getName()
Adding s2_to_dsm return for runStage2()

* Tuning up TpcRS for AuAu19GeV 2019 (#184)

* Update CHANGELOG for SL21d_0 release (#192)

[skip ci]

* Star tablelib (#189)

* Cleanup for ROOT 6.

Explicitly include header files which we depend on / don't assume namespace std.

* The libTable library was deprecated following ROOT 6.16.  In order to support
our framework with later versions of ROOT, we (re) import the libTable sources
into our repository.

In order to support building our software stack with versions of ROOT which may
provide support for libTable, we modify cons to detect whether or not libTable
support was compiled in.  If ROOT provides libTable, we do not build it and
do not export header files into $STAR_HOST_SYS/include.  If ROOT does not provide
libTable, we build it ourselves.

* Additional ROOT 6 cleanup.

* ci: Avoid conflict with latest tag when using cache to/from (#198)

Co-authored-by: Dmitry Kalinkin <dmitry.kalinkin@gmail.com>

* Fix cut limits in EPD macro (#188)

* Fixed bugs by MAL

* Apply suggestions from code review

Co-authored-by: Dmitry Kalinkin <dmitry.kalinkin@gmail.com>

* Fcs mudst (#200)

* Initial readme on how to sparse checkout StarVMC

* add .gitignore for working directories StarGeometry and xgeometry

* edit description comment

* Rotated middle disk by 30 deg to mactch actual installation.

* simplified base file for fst geometry
Updated StarGeo.xml to be able to acces this in dev2021

* Simplified FST geometry v1.1 ( cooling tubes, ribs, cooling tube holder, heatsinks (draft) )

* Adjustments to commented out simplified heatsinks and included 30 deg rotation of middle disk

* Fixed syntax issue with middle disk rotation

* Updated material definition for combined stainless and novec

* Added 3 more cooling tubes, compressed heatsinks into single volumes in PEEK bases, removed unnecessary code

* Compressed heatsinks and added 3 additional cooling tube volumes (simplified cooling tube connectors)

* accept gavins changes. removed some dummy not used lines

* Ring removed.

* remove README for pull request

* Fixes for PR

* Initial addition of Fcs to MuDst, with StEvent->MuDst and MuDst->StEvent support

* Addition of StMuFcsInfo, store reconstruction flag and detector indices, add helper methods to StMuFcsCollection

* Fix old comments for FCS classes

Co-authored-by: Flemming Videbaek <videbaks@rcas6010.rcf.bnl.gov>
Co-authored-by: Flemming Videbaek <videbaek@bnl.gov>
Co-authored-by: gavin-wilks <gavin.wilks@gmail.com>
Co-authored-by: Dmitri Smirnov <dmixsmi@gmail.com>

* updates to daq readers in prep for run 22 (#186)

* Remove StRoot/RTS/include/FTP/FTPC_PADKEY.h (#195)

This file appears to be unused but conflicts with StRoot/RTS/include/FTP/ftpc_padkey.h

Removing it resolves #194. For details see https://github.com/star-bnl/star-sw/issues/194

* StEvent StTriggerData2022 update associate with  (#203)

* Removing some EPD DSM from trigger structure
StEvent StTriggerData2022 update associate with those removal

* Leaving removed EPD DSMs in place for now, Just adding comments

* Trigger group removed 5DSM from BBC crate, thus changing structure. (#207)

* StRoot/Stv*: replace sys/types.h with stdint.h (#211)

* StEvent additions for sTGC (Ftt) detector (#209)

* Initial readme on how to sparse checkout StarVMC

* add .gitignore for working directories StarGeometry and xgeometry

* edit description comment

* Rotated middle disk by 30 deg to mactch actual installation.

* simplified base file for fst geometry
Updated StarGeo.xml to be able to acces this in dev2021

* Simplified FST geometry v1.1 ( cooling tubes, ribs, cooling tube holder, heatsinks (draft) )

* Adjustments to commented out simplified heatsinks and included 30 deg rotation of middle disk

* Fixed syntax issue with middle disk rotation

* Updated material definition for combined stainless and novec

* Added 3 more cooling tubes, compressed heatsinks into single volumes in PEEK bases, removed unnecessary code

* Compressed heatsinks and added 3 additional cooling tube volumes (simplified cooling tube connectors)

* accept gavins changes. removed some dummy not used lines

* Ring removed.

* remove README for pull request

* Fixes for PR

* StEvent additions for sTGC (Ftt) detector. Including inital version of hits, clusters, and points

* Add StFttRawHitMaker and BFC option for running it (FttDat)

* change a few LOG_INFO -> LOG_DEBUG

* Remove empty dtor from StFttRawHitMaker, include RTS daq_stgc with CINT guard

Co-authored-by: Flemming Videbaek <videbaks@rcas6010.rcf.bnl.gov>
Co-authored-by: Flemming Videbaek <videbaek@bnl.gov>
Co-authored-by: gavin-wilks <gavin.wilks@gmail.com>
Co-authored-by: Dmitri Smirnov <dmixsmi@gmail.com>

* Initial Run 22 chains, including FTT and FCS (#217)

* StFcsDB makes no file unless debug is on (#218)

Crash fixed in StMuFcsUtil::fillMuFcsHit()

* Minor fix: 2017 -> 2022 (#219)

* Move back to MagFactor instead of starMagOnl for pre 2012 data (issue #185) (#216)

* Create README.md

* Update README.md

* Move back to use  MagFactor for data obtained before 2012 (before starMagAvg), due to missing run numbers used for RunLog/onl DB. Use fixed scale to 4500 kA for starMagOnl

* Delete README.md

Co-authored-by: Dmitri Smirnov <dmixsmi@gmail.com>

* Fix min/max default value mixup (#221)

* StMuDstMaker, init mFcsCollection to zero (#222)

* StRoot/StTriggerUtilities: replace sys/types.h with stdint.h (#212)

* starsim: make seed overflow into a fatal error (#215)

The value of bigger than 0x7FFFFFFF will overflow ISEQ and make it
wrap around into a negative values. The current behavior is to print a
warning message and reset the seed to a default value. Because of that
an unsuspecting user can easily initialize a fraction of their
simulation production with the same seed.

* HOTFIX: Missing FttCollection in StEventClusteringHints.cxx (#227)

* Initial readme on how to sparse checkout StarVMC

* add .gitignore for working directories StarGeometry and xgeometry

* edit description comment

* Rotated middle disk by 30 deg to mactch actual installation.

* simplified base file for fst geometry
Updated StarGeo.xml to be able to acces this in dev2021

* Simplified FST geometry v1.1 ( cooling tubes, ribs, cooling tube holder, heatsinks (draft) )

* Adjustments to commented out simplified heatsinks and included 30 deg rotation of middle disk

* Fixed syntax issue with middle disk rotation

* Updated material definition for combined stainless and novec

* Added 3 more cooling tubes, compressed heatsinks into single volumes in PEEK bases, removed unnecessary code

* Compressed heatsinks and added 3 additional cooling tube volumes (simplified cooling tube connectors)

* accept gavins changes. removed some dummy not used lines

* Ring removed.

* remove README for pull request

* Fixes for PR

* hotfix: missing FttCollection in StEventClusteringHints.cxx

Co-authored-by: Flemming Videbaek <videbaks@rcas6010.rcf.bnl.gov>
Co-authored-by: Flemming Videbaek <videbaek@bnl.gov>
Co-authored-by: gavin-wilks <gavin.wilks@gmail.com>
Co-authored-by: Dmitri Smirnov <dmixsmi@gmail.com>

* StRoot/StSpinPool: replace sys/types.h with stdint.h (#224)

* Update gstar_part.g (#228)

updated the K0*(892) mass and width according to the latest PDG values.

* Adding option to read from MuDst in StFcsRawHitMaker (#226)

* Adding option to read from Mudst in StFcsRawHitMaker, and example macro to read from Mudst, re-create StFcsHit in StEvent on momery, and re-run FCS reconstruction makers.

* unsigned int -> int

* update fstBuilder for beam

* update fstBuilder for Lat Tuning

* ftt builder's fitTriggerTime must be protected with mutex

* update to fcsBuilder

* trgBuilder

* updated again

* updates to trgBuilder

* add trgBuilder plots, and fix a bug

* update for 3 time bin

* set logy for ZDCSMD adc plots

* Update fttBuilder.h

Add new plot for timebin zoomed in

* Update fttBuilder.cxx

Update nStripsHit histogram,
add zoomed in timebin fit histogram

* update bbcBuilder

* Do not fill MTD histograms if MTD is not read out

* added various logging

* updated logging

* set default minimum of 4 orders of magitude less than max for logy on log plots

* fix tofBuilder

* tofBuilder update

* split tpc builder to add laser builder

* update for TH1 arrays in unions for c++11

* Create PWGTools area (#340)

The commit will create a new directory that should be excluded from the
compilation process. At the first iteration it should be cross checked
(if compilation will pass the CI/CD). Also before merging Zachary Sweger
should confirm that the subdirectories contain the latest versions of
the Glauber codes.

---------

Co-authored-by: Grigory Nigmatkulov <nigmaktulov@gmail.com>
Co-authored-by: Dmitri Smirnov <dmixsmi@gmail.com>
Co-authored-by: Zachary Sweger <zsweger@berkeley.edu>

* Summer 2023 updates

* update level.source

* adding DAQ_TPC23 directory

* remove root files

* fix itpc23.h

* new directions in readme.txt

* tonko updates

* remove logging from itpcFCF and local daq updates

* fix new cluster finder dropping outer two RDOs of TPX bug

* fix laser reader

* updates

* added blue/yellow beam by trigger plots

* updates to fst/laser

* Changed Vertex Z plotting range for Fix Target and Fix Target Monitor
from (190,210) to (-190,-170) in l4Builder.cxx

* update to maxtac in vpd builder

* DAQ reader updates for DAQ5k

* update plots for time bin range and mult tb cut

* fix fstBuilder for 9 timebins, and commit earlier changes to fttBuilder

* updates to vpd builder

* updated with newest readers

---------

Co-authored-by: Jeff Landgraf <jml@evp.starp.bnl.gov>
Co-authored-by: jml985 <jml985@yahoo.com>
Co-authored-by: jml <jml@bnl.gov>
Co-authored-by: jml985 <44065529+jml985@users.noreply.github.com>
Co-authored-by: tonko-lj <tonko.ljubicic@gmail.com>
Co-authored-by: Event Pool and Online Plots <evpops@evp.starp.bnl.gov>
Co-authored-by: Xu Sun <sunxuhit@gmail.com>
Co-authored-by: Dmitri Smirnov <dmixsmi@gmail.com>
Co-authored-by: Akio Ogawa <akio@bnl.gov>
Co-authored-by: jdb12@rice.edu <jdb12@rice.edu>
Co-authored-by: RongrongMa <marr@bnl.gov>
Co-authored-by: Yuri Fisyak <fisyak@bnl.gov>
Co-authored-by: klendathu2k <56083924+klendathu2k@users.noreply.github.com>
Co-authored-by: Dmitry Arkhipkin <arkhipkin@gmail.com>
Co-authored-by: Dmitry Arkhipkin <dmitry@localhost.localdomain>
Co-authored-by: Gene Van Buren <85305093+genevb@users.noreply.github.com>
Co-authored-by: akioogawa <85249930+akioogawa@users.noreply.github.com>
Co-authored-by: PhilippWeidenkaff <36641002+PhilippWeidenkaff@users.noreply.github.com>
Co-authored-by: Philipp Weidenkaff <weidenkaff@rcas6005.rcf.bnl.gov>
Co-authored-by: PhilippWeidenkaff <weidenkaff@physi.uni-heildeberg.de>
Co-authored-by: Dmitry Kalinkin <dmitry.kalinkin@gmail.com>
Co-authored-by: liuxiaoyuyuyu <44676148+liuxiaoyuyuyu@users.noreply.github.com>
Co-authored-by: Daniel Brandenburg <danbburg@gmail.com>
Co-authored-by: Flemming Videbaek <videbaks@rcas6010.rcf.bnl.gov>
Co-authored-by: Flemming Videbaek <videbaek@bnl.gov>
Co-authored-by: gavin-wilks <gavin.wilks@gmail.com>
Co-authored-by: zhux97 <86642699+zhux97@users.noreply.github.com>
Co-authored-by: Hongwei Ke <kehw@bnl.gov>
Co-authored-by: nigmatkulov <nigmatkulov@gmail.com>
Co-authored-by: Grigory Nigmatkulov <nigmaktulov@gmail.com>
Co-authored-by: Zachary Sweger <zsweger@berkeley.edu>
Co-authored-by: Diyu Shen <dyshen@rcf.rhic.bnl.gov>

zyzhang-0 committed:
1. Fix a bug in frac vs apvidx on histogram bin index \n 2.Slightly change the definition of fraction denominators (won't affect 3 time bin) \n 3. Define control variable of plot_tb_low and plot_tb_high; utilize them to change the plotting range for Time Bin plots (#2)

Co-authored-by: jml985 <jml@bnl.gov>
